main.go

```diff
@@ -1547,6 +1560,159 @@ func (c MainController) ListAllChapters(w http.ResponseWriter, r *http.Request)
 	writeJSON(w, chapters)
 }
 
+// testing this
+func (c MainController) discordBotAuthMiddleware(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		err := r.ParseForm()
+		if err != nil {
+			http.Error(w, http.StatusText(400), 400)
+			return
+		}
+
+		// check that discord auth matches (shared secret w/ discord bot)
+		discordAuth := r.PostFormValue("auth")
+		if discordAuth != config.DiscordSecret {
```

Comment:
At least theoretically, this is vulnerable to a timing attack. Might as well use  `subtle.ConstantTimeCompare([]byte(discordAuth), []byte(config.DiscordSecret)) != 1` instead though.